core/src/main/scala/cats/syntax/applicative.scala

```diff
@@ -17,3 +19,8 @@ final class ApplicativeOps[F[_], A](private val fa: F[A]) extends AnyVal {
   def unlessA(cond: Boolean)(implicit F: Applicative[F]): F[Unit] = F.unlessA(cond)(fa)
   def whenA(cond: Boolean)(implicit F: Applicative[F]): F[Unit] = F.whenA(cond)(fa)
 }
+
+final class ApplicativeByNameOps[F[_], A](private val fa: () => F[A]) extends AnyVal {
+  def unlessA_(cond: Boolean)(implicit F: Applicative[F]): F[Unit] = F.unlessA(cond)(fa())
```

Comment:
I think we should use .void on these two.

Reply:
Hmm, what for?

Reply:
sorry, I got confused. Often we have `foo_` as a variant of `foo` that discards the value. But I see that here we are doing something else (just using it as another version that uses call by name correctly.

I would like to see another name since, I think, that pattern is established and this would be a violation of it.